src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+using Microsoft.VisualStudio.Services.CircuitBreaker;
+using Microsoft.WorkItems;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public class SarifWorkItemFiler
+    {
+        public SarifWorkItemFiler() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SarifWorkItemFiler"> class.</see>
+        /// </summary>
+        /// <param name="filingClient">
+        /// A client for communicating with a work item filing host (for example, GitHub or Azure DevOps).
+        /// </param>
+        /// <param name="filingContext">
+        /// A root context object that configures the work item filing operation.
```

Comment:
>root [](start = 14, length = 4)

"root" -- are there also "child" objects?

Reply:
Ah, the comment on L135 explains it. But these XML doc comments are potentially read from documentation pages where the source code is not present, and even if the code is present, it takes a while to understand the significance of this word. I'd just say "A context object". The person who wants to understand this method doesn't care that later on this context will be adapted for each of the split logs.

---
In reply to: [381459277](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381459277) [](ancestors = 381459277)

Reply:
Updated the comment to clarify this is 'starting' context which is copied for any split SARIF log files (if any).

---
In reply to: [381481907](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381481907) [](ancestors = 381481907,381459277)